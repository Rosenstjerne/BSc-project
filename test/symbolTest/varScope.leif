var int x;
function int test_scope() {
    var int x;
    x = 10;
    return x;
}

x = 5;
print(test_scope()); # Should print 10
print(x); # Should print 5
