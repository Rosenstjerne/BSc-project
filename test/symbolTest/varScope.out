function null main() {
	var int x;
	
	function int test_scope() {
		var int x;
		
		x = 10;
		return x;
		
	}
	
	x = 5;
	print(test_scope());
	print(x);
	
}
